app/controllers/clients_controller.rb

```diff
@@ -6,7 +6,10 @@ class ClientsController < ApplicationController
   before_action :set_client, only: [:show, :edit, :update, :destroy]
 
   def index
-    @clients = policy_scope(Client).paginate(page: params[:page])
+    authorize Client
+    @clients = ClientPolicy::Scope.new(
+      current_user, Client
+    ).resolve_superadmin_or_social_worker_owns.paginate(page: params[:page])
```

Comment:
I tryed to use methods inherited from ApplicationPolicy, instead of having them inside the controller specific policies. But not sure if that is very stilish to do it in that manner.